Build server crashes with Gradle v8.9

Extension Name: vscode-gradle
Extension Version: `v3.14.1`
OS Version: Windows 11 (Windows_NT x64 10.0.22631)
VSCode version: `v1.91.1`
Language Support for Java: `v1.32.0`

**Describe the bug**
The build server worked fine until Gradle v8.8, but once I updated my Gradle Wrapper to v8.9, the build server crashed, causing the Java Language Server to fail to open the Java project. However, if I set `java.gradle.buildServer.enabled: off`, the project loads with no problems.

**To Reproduce**
- Open a Gradle project using Gradle Wrapper v8.9.
- Wait for the Java Status badge to show `Java: Error`

**Expected behavior**
The build server should work on Gradle v8.9

**Screenshots**
![image](https://github.com/user-attachments/assets/bcae9c89-bdff-4080-bfce-3086c106748b)

**Output from "Gradle for Java"**
```shell
[info] Build file changed: c:\Repos\maybe\build.gradle
[info] Java Home: C:\Program Files\Java\jdk-22
[info] JVM Args: --add-opens=java.base/java.util=ALL-UNNAMED,--add-opens=java.base/java.lang=ALL-UNNAMED,--add-opens=java.base/java.lang.invoke=ALL-UNNAMED,--add-opens=java.prefs/java.util.prefs=ALL-UNNAMED,--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED,--add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED,--add-opens=java.base/java.nio.charset=ALL-UNNAMED,--add-opens=java.base/java.net=ALL-UNNAMED,--add-opens=java.base/java.util.concurrent.atomic=ALL-UNNAMED,-XX:MaxMetaspaceSize=384m,-XX:+HeapDumpOnOutOfMemoryError,-Xms256m,-Xmx512m,-Dfile.encoding=UTF-8,-Duser.country=US,-Duser.language=en,-Duser.variant
[info] Gradle User Home: C:\Users\josel\.gradle
[info] Gradle Version: 8.9
[info] Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.9/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

CONFIGURE SUCCESSFUL in 285ms
[info] Found 83 tasks
```

**Does the bug still exist if you disable all other extensions?**
Yes

**Additional context**

For some reason, the `Gradle For Java` and `Language Support for Java(TM) by Red Hat` extensions automatically update to their pre-release versions. The issue problem persists in the pre-release versions:
  - Language Support for Java: `v1.33.2024062908 (pre-release)`
  - Gradle for Java: `v3.14.2024071106 (pre-release)`

Because the Java Language Server also fails to initialize, the following logs can also be found:

<details>
<summary><i>redhat.java/jdt_ws/.metadata/.log</i></summary>

```shell
!SESSION 2024-07-13 14:14:23.743 -----------------------------------------------
eclipse.buildId=unknown
java.version=17.0.11
java.vendor=Eclipse Adoptium
BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=en_US
Framework arguments:  --pipe=\\.\pipe\lsp-d8ce395af0525274e1a35f233e5fa225-sock
Command-line arguments:  -data c:\Users\josel\AppData\Roaming\Code\User\workspaceStorage\8778ce053cead4dbffa3405251da4ae0\redhat.java\jdt_ws --pipe=\\.\pipe\lsp-d8ce395af0525274e1a35f233e5fa225-sock

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:25.747
!MESSAGE class org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin is started

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:25.995
!MESSAGE Main thread is waiting

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:26.026
!MESSAGE >> initialize

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:26.032
!MESSAGE Initializing Java Language Server 1.37.0.202406270241

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:26.035
!MESSAGE Started org.eclipse.m2e.core 0ms

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:26.776
!MESSAGE ProjectRegistryRefreshJob finished 739ms

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:26.804
!MESSAGE Started org.eclipse.buildship.core 24ms

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:26.912
!MESSAGE Setting C:\Program Files\Java\jdk-19 as 'JavaSE-19' environment (id:1720898066893)

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:26.930
!MESSAGE Setting C:\Program Files\Java\jdk-21 as 'JavaSE-21' environment (id:1720898066919)

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:26.944
!MESSAGE Setting C:\Program Files\Java\jdk-20 as 'JavaSE-20' environment (id:1720898066935)

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:26.956
!MESSAGE Setting C:\Program Files\Java\jdk-22 as 'JavaSE-22' environment (id:1720898066948)

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:26.967
!MESSAGE Setting C:\Program Files\Java\jdk-11 as 'JavaSE-11' environment (id:1720898066959)

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:26.979
!MESSAGE Setting C:\Program Files\Java\jdk-18.0.2.1 as 'JavaSE-18' environment (id:1720898066971)

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:26.991
!MESSAGE Setting C:\Users\josel\.gradle\jdks\eclipse_adoptium-17-amd64-windows\jdk-17.0.11+9 as 'JavaSE-17' environment (id:1720898066983)

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:26.995
!MESSAGE Setting java.home C:\Program Files\Java\jdk-22 as default global VM

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.008
!MESSAGE Default VM Install changed from  1720898066869-C:\Users\josel\.vscode\extensions\redhat.java-1.32.0-win32-x64\jre\17.0.11-win32-x86_64 to 1720898066948-C:\Program Files\Java\jdk-22

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.030
!MESSAGE JVM Runtimes changed, saving new configuration

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.054
!MESSAGE Static Commands: []

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.057
!MESSAGE Non-Static Commands: [java.project.import, java.project.changeImportedProjects, java.navigate.openTypeHierarchy, java.project.resolveStackTraceLocation, java.edit.handlePasteEvent, java.edit.stringFormatting, java.project.getSettings, java.project.resolveWorkspaceSymbol, java.project.upgradeGradle, java.project.createModuleInfo, java.vm.getAllInstalls, java.edit.organizeImports, java.project.refreshDiagnostics, java.project.removeFromSourcePath, java.project.listSourcePaths, java.project.updateSettings, java.project.getAll, java.reloadBundles, java.project.isTestFile, java.project.resolveText, java.project.getClasspaths, java.navigate.resolveTypeHierarchy, java.edit.smartSemicolonDetection, java.project.updateSourceAttachment, java.project.updateClassPaths, java.decompile, java.protobuf.generateSources, java.project.resolveSourceAttachment, java.project.updateJdk, java.project.addToSourcePath, java.completion.onDidSelect]

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.058
!MESSAGE Static Commands: []

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.060
!MESSAGE Non-Static Commands: [java.checkstyle.server.checkCode, java.checkstyle.server.setConfiguration, java.checkstyle.server.getVersion, java.checkstyle.server.quickFix]

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.061
!MESSAGE Static Commands: []

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.062
!MESSAGE Non-Static Commands: [java.intellicode.enable]

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.064
!MESSAGE Static Commands: [sts.java.addClasspathListener, sts.java.removeClasspathListener]

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.065
!MESSAGE Non-Static Commands: []

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.068
!MESSAGE Static Commands: []

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.069
!MESSAGE Non-Static Commands: [sts.java.javadoc]

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.071
!MESSAGE Static Commands: [sts.java.type]

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.072
!MESSAGE Non-Static Commands: []

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.074
!MESSAGE Static Commands: []

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.077
!MESSAGE Non-Static Commands: [sts.java.javadocHoverLink]

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.079
!MESSAGE Static Commands: []

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.081
!MESSAGE Non-Static Commands: [sts.java.location]

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.085
!MESSAGE Static Commands: [sts.java.search.packages, sts.java.search.types]

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.086
!MESSAGE Non-Static Commands: []

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.089
!MESSAGE Static Commands: [sts.java.hierarchy.supertypes, sts.java.hierarchy.subtypes]

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.091
!MESSAGE Non-Static Commands: []

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.095
!MESSAGE Static Commands: []

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.097
!MESSAGE Non-Static Commands: [sts.java.code.completions]

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.100
!MESSAGE Static Commands: []

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.103
!MESSAGE Non-Static Commands: [sts.project.gav]

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.105
!MESSAGE Static Commands: []

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.107
!MESSAGE Non-Static Commands: [vscode.java.checkProjectSettings, vscode.java.isOnClasspath, vscode.java.fetchUsageData, vscode.java.validateLaunchConfig, vscode.java.resolveInlineVariables, vscode.java.resolveClassFilters, vscode.java.resolveMainMethod, vscode.java.resolveClasspath, vscode.java.resolveBuildFiles, vscode.java.resolveMainClass, vscode.java.updateDebugSettings, vscode.java.resolveSourceUri, vscode.java.fetchPlatformSettings, vscode.java.buildWorkspace, vscode.java.startDebugSession, vscode.java.inferLaunchCommandLength, vscode.java.resolveElementAtSelection, vscode.java.resolveJavaExecutable]

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.110
!MESSAGE Static Commands: []

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.113
!MESSAGE Non-Static Commands: [java.project.refreshLib, java.project.checkImportStatus, java.project.list, java.project.generateJar, java.project.getMainClasses, java.getPackageData, java.resolvePath]

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.116
!MESSAGE Static Commands: []

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.119
!MESSAGE Non-Static Commands: [vscode.java.test.jacoco.getCoverageDetail, vscode.java.test.findJavaProjects, vscode.java.test.findTestPackagesAndTypes, vscode.java.test.findTestTypesAndMethods, vscode.java.test.resolvePath, vscode.java.test.findTestLocation, vscode.java.test.get.testpath, vscode.java.test.findDirectTestChildrenForClass, vscode.java.test.navigateToTestOrTarget, vscode.java.test.junit.argument, vscode.java.test.generateTests]

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.122
!MESSAGE Static Commands: []

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.124
!MESSAGE Non-Static Commands: [java.codeAction.lombok.getAnnotations, java.codeAction.lombok]

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.125
!MESSAGE Static Commands: []

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.128
!MESSAGE Non-Static Commands: [java.maven.initializeSearcher, java.maven.searchArtifact, java.maven.addDependency, java.maven.controlContext]

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.143
!MESSAGE RepositoryRegistryUpdateJob finished 0ms

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:27.149
!MESSAGE >> initialized

!ENTRY org.eclipse.jdt.ls.core 4 0 2024-07-13 14:14:28.563
!MESSAGE Initialization failed 
!STACK 0
java.util.concurrent.CompletionException: org.eclipse.lsp4j.jsonrpc.ResponseErrorException: org.gradle.tooling.exceptions.UnsupportedBuildArgumentException: Could not fetch model of type 'BuildEnvironment' using connection to Gradle distribution 'https://services.gradle.org/distributions/gradle-8.9-bin.zip'.
Problem with provided build arguments: [--no-daemon]. 
Unknown command-line option '--no-daemon'.
Either it is not a valid build option or it is not supported in the target Gradle version.
Not all of the Gradle command line options are supported build arguments.
Examples of supported build arguments: '--info', '-p'.
Examples of unsupported build options: '--daemon', '-?', '-v'.
Please find more information in the javadoc for the BuildLauncher class.
	at java.base/java.util.concurrent.CompletableFuture.reportJoin(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.join(Unknown Source)
	at com.microsoft.gradle.bs.importer.GradleBuildServerProjectImporter.importToWorkspace(GradleBuildServerProjectImporter.java:171)
	at org.eclipse.jdt.ls.core.internal.managers.ProjectsManager.importProjects(ProjectsManager.java:164)
	at org.eclipse.jdt.ls.core.internal.managers.ProjectsManager.initializeProjects(ProjectsManager.java:126)
	at org.eclipse.jdt.ls.core.internal.handlers.InitHandler$1.runInWorkspace(InitHandler.java:277)
	at org.eclipse.core.internal.resources.InternalWorkspaceJob.run(InternalWorkspaceJob.java:43)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: org.eclipse.lsp4j.jsonrpc.ResponseErrorException: org.gradle.tooling.exceptions.UnsupportedBuildArgumentException: Could not fetch model of type 'BuildEnvironment' using connection to Gradle distribution 'https://services.gradle.org/distributions/gradle-8.9-bin.zip'.
Problem with provided build arguments: [--no-daemon]. 
Unknown command-line option '--no-daemon'.
Either it is not a valid build option or it is not supported in the target Gradle version.
Not all of the Gradle command line options are supported build arguments.
Examples of supported build arguments: '--info', '-p'.
Examples of unsupported build options: '--daemon', '-?', '-v'.
Please find more information in the javadoc for the BuildLauncher class.
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleResponse(RemoteEndpoint.java:220)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:204)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:28.583
!MESSAGE >> initialization job finished

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:28.629
!MESSAGE >> build jobs finished

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:28.632
!MESSAGE >> registerWatchers'

!ENTRY org.eclipse.jdt.ls.core 1 0 2024-07-13 14:14:28.638
!MESSAGE >> registerFeature 'workspace/didChangeWatchedFiles'
```

</details>

<details>
<summary><i>redhat.java/client.log.2024-07-13</i></summary>

```shell
{
  message: "Use the JDK from 'JAVA_HOME,PATH' as the initial default project JDK.",
  level: 'info',
  timestamp: '2024-07-13 14:14:22.644'
}
{
  message: 'Starting Java server with: c:\\Users\\josel\\.vscode\\extensions\\redhat.java-1.32.0-win32-x64\\jre\\17.0.11-win32-x86_64\\bin\\java --add-modules=ALL-SYSTEM --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/sun.nio.fs=ALL-UNNAMED -Declipse.application=org.eclipse.jdt.ls.core.id1 -Dosgi.bundles.defaultStartLevel=4 -Declipse.product=org.eclipse.jdt.ls.core.product -Djava.import.generatesMetadataFilesAtProjectRoot=false -DDetectVMInstallationsJob.disabled=true -Xlog:disable -XX:+UseParallelGC -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -Dsun.zip.disableMemoryMapping=true -Xmx4G -Xms100m -Dfile.encoding=UTF-8 -javaagent:c:\\Users\\josel\\.vscode\\extensions\\redhat.java-1.32.0-win32-x64\\lombok\\lombok-1.18.33.jar -jar c:\\Users\\josel\\.vscode\\extensions\\redhat.java-1.32.0-win32-x64\\server\\plugins\\org.eclipse.equinox.launcher_1.6.900.v20240613-2009.jar -configuration c:\\Users\\josel\\AppData\\Roaming\\Code\\User\\globalStorage\\redhat.java\\1.32.0\\config_ss_win -data c:\\Users\\josel\\AppData\\Roaming\\Code\\User\\workspaceStorage\\8778ce053cead4dbffa3405251da4ae0\\redhat.java\\ss_ws',
  level: 'info',
  timestamp: '2024-07-13 14:14:22.850'
}
{
  message: 'Starting Java server with: c:\\Users\\josel\\.vscode\\extensions\\redhat.java-1.32.0-win32-x64\\jre\\17.0.11-win32-x86_64\\bin\\java --add-modules=ALL-SYSTEM --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/sun.nio.fs=ALL-UNNAMED -Declipse.application=org.eclipse.jdt.ls.core.id1 -Dosgi.bundles.defaultStartLevel=4 -Declipse.product=org.eclipse.jdt.ls.core.product -Djava.import.generatesMetadataFilesAtProjectRoot=false -DDetectVMInstallationsJob.disabled=true -Xlog:disable -XX:+UseParallelGC -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -Dsun.zip.disableMemoryMapping=true -Xmx4G -Xms100m -Dfile.encoding=UTF-8 -javaagent:c:\\Users\\josel\\.vscode\\extensions\\redhat.java-1.32.0-win32-x64\\lombok\\lombok-1.18.33.jar -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=c:\\Users\\josel\\AppData\\Roaming\\Code\\User\\workspaceStorage\\8778ce053cead4dbffa3405251da4ae0\\redhat.java -Daether.dependencyCollector.impl=bf -jar c:\\Users\\josel\\.vscode\\extensions\\redhat.java-1.32.0-win32-x64\\server\\plugins\\org.eclipse.equinox.launcher_1.6.900.v20240613-2009.jar -configuration c:\\Users\\josel\\AppData\\Roaming\\Code\\User\\globalStorage\\redhat.java\\1.32.0\\config_win -data c:\\Users\\josel\\AppData\\Roaming\\Code\\User\\workspaceStorage\\8778ce053cead4dbffa3405251da4ae0\\redhat.java\\jdt_ws',
  level: 'info',
  timestamp: '2024-07-13 14:14:23.271'
}
{
  message: 'Jul 13, 2024 2:14:24 PM org.apache.aries.spifly.BaseActivator log\r\n' +
    'INFO: Registered provider ch.qos.logback.classic.servlet.LogbackServletContainerInitializer of service jakarta.servlet.ServletContainerInitializer in bundle ch.qos.logback.classic\r\n',
  level: 'info',
  timestamp: '2024-07-13 14:14:24.327'
}
{
  message: 'Jul 13, 2024 2:14:24 PM org.apache.aries.spifly.BaseActivator log\r\n' +
    'INFO: Registered provider ch.qos.logback.classic.spi.LogbackServiceProvider of service org.slf4j.spi.SLF4JServiceProvider in bundle ch.qos.logback.classic\r\n',
  level: 'info',
  timestamp: '2024-07-13 14:14:24.330'
}
{
  message: 'Jul 13, 2024 2:14:28 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint handleNotification\r\n' +
    'WARNING: Notification threw an exception: {\n' +
    '  "jsonrpc": "2.0",\n' +
    '  "method": "build/logMessage",\n' +
    '  "params": {\n' +
    '    "type": 1,\n' +
    '    "message": "[SEVERE - 14:14:28] Failed to process \\u0027build/initialize\\u0027: java.util.concurrent.CompletionException: org.gradle.tooling.exceptions.UnsupportedBuildArgumentException: Could not fetch model of type \\u0027BuildEnvironment\\u0027 using connection to Gradle distribution \\u0027https://services.gradle.org/distributions/gradle-8.9-bin.zip\\u0027.\\nProblem with provided build arguments: [--no-daemon]. \\nUnknown command-line option \\u0027--no-daemon\\u0027.\\nEither it is not a valid build option or it is not supported in the target Gradle version.\\nNot all of the Gradle command line options are supported build arguments.\\nExamples of supported build arguments: \\u0027--info\\u0027, \\u0027-p\\u0027.\\nExamples of unsupported build options: \\u0027--daemon\\u0027, \\u0027-?\\u0027, \\u0027-v\\u0027.\\nPlease find more information in the javadoc for the BuildLauncher class.\\r\\n\\tat java.base/java.util.concurrent.CompletableFuture.encodeThrowable(Unknown Source)\\r\\n\\tat java.base/java.util.concurrent.CompletableFuture.completeThrowable(Unknown Source)\\r\\n\\tat java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(Unknown Source)\\r\\n\\tat java.base/java.util.concurrent.CompletableFuture$Completion.exec(Unknown Source)\\r\\n\\tat java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)\\r\\n\\tat java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)\\r\\n\\tat java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)\\r\\n\\tat java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)\\r\\n\\tat java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)\\r\\nCaused by: org.gradle.tooling.exceptions.UnsupportedBuildArgumentException: Could not fetch model of type \\u0027BuildEnvironment\\u0027 using connection to Gradle distribution \\u0027https://services.gradle.org/distributions/gradle-8.9-bin.zip\\u0027.\\nProblem with provided build arguments: [--no-daemon]. \\nUnknown command-line option \\u0027--no-daemon\\u0027.\\nEither it is not a valid build option or it is not supported in the target Gradle version.\\nNot all of the Gradle command line options are supported build arguments.\\nExamples of supported build arguments: \\u0027--info\\u0027, \\u0027-p\\u0027.\\nExamples of unsupported build options: \\u0027--daemon\\u0027, \\u0027-?\\u0027, \\u0027-v\\u0027.\\nPlease find more information in the javadoc for the BuildLauncher class.\\r\\n\\tat org.gradle.tooling.internal.consumer.ExceptionTransformer.transform(ExceptionTransformer.java:40)\\r\\n\\tat org.gradle.tooling.internal.consumer.ExceptionTransformer.transform(ExceptionTransformer.java:29)\\r\\n\\tat org.gradle.tooling.internal.consumer.ResultHandlerAdapter.onFailure(ResultHandlerAdapter.java:43)\\r\\n\\tat org.gradle.tooling.internal.consumer.async.DefaultAsyncConsumerActionExecutor$1$1.run(DefaultAsyncConsumerActionExecutor.java:69)\\r\\n\\tat org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)\\r\\n\\tat org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:47)\\r\\n\\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)\\r\\n\\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)\\r\\n\\tat java.base/java.lang.Thread.run(Unknown Source)\\r\\n\\tat org.gradle.tooling.internal.consumer.BlockingResultHandler.getResult(BlockingResultHandler.java:46)\\r\\n\\tat org.gradle.tooling.internal.consumer.DefaultModelBuilder.get(DefaultModelBuilder.java:51)\\r\\n\\tat com.microsoft.java.bs.core.internal.gradle.GradleApiConnector.getGradleVersion(GradleApiConnector.java:53)\\r\\n\\tat com.microsoft.java.bs.core.internal.services.LifecycleService.updateGradleJavaHomeIfNecessary(LifecycleService.java:148)\\r\\n\\tat com.microsoft.java.bs.core.internal.services.LifecycleService.initializePreferenceManager(LifecycleService.java:80)\\r\\n\\tat com.microsoft.java.bs.core.internal.services.LifecycleService.initializeServer(LifecycleService.java:57)\\r\\n\\tat com.microsoft.java.bs.core.internal.server.GradleBuildServer.lambda$buildInitialize$0(GradleBuildServer.java:80)\\r\\n\\t... 7 more\\r\\nCaused by: org.gradle.tooling.internal.protocol.exceptions.InternalUnsupportedBuildArgumentException: Problem with provided build arguments: [--no-daemon]. \\nUnknown command-line option \\u0027--no-daemon\\u0027.\\nEither it is not a valid build option or it is not supported in the target Gradle version.\\nNot all of the Gradle command line options are supported build arguments.\\nExamples of supported build arguments: \\u0027--info\\u0027, \\u0027-p\\u0027.\\nExamples of unsupported build options: \\u0027--daemon\\u0027, \\u0027-?\\u0027, \\u0027-v\\u0027.\\nPlease find more information in the javadoc for the BuildLauncher class.\\r\\n\\tat org.gradle.tooling.internal.provider.ProviderStartParameterConverter.toStartParameter(ProviderStartParameterConverter.java:81)\\r\\n\\tat org.gradle.tooling.internal.provider.ProviderConnection.initParams(ProviderConnection.java:384)\\r\\n\\tat org.gradle.tooling.internal.provider.ProviderConnection.run(ProviderConnection.java:155)\\r\\n\\tat org.gradle.tooling.internal.provider.DefaultConnection.getModel(DefaultConnection.java:153)\\r\\n\\tat org.gradle.tooling.internal.consumer.connection.CancellableModelBuilderBackedModelProducer.produceModel(CancellableModelBuilderBackedModelProducer.java:54)\\r\\n\\tat org.gradle.tooling.internal.consumer.connection.PluginClasspathInjectionSupportedCheckModelProducer.produceModel(PluginClasspathInjectionSupportedCheckModelProducer.java:38)\\r\\n\\tat org.gradle.tooling.internal.consumer.connection.AbstractConsumerConnection.run(AbstractConsumerConnection.java:64)\\r\\n\\tat org.gradle.tooling.internal.consumer.connection.ParameterValidatingConsumerConnection.run(ParameterValidatingConsumerConnection.java:49)\\r\\n\\tat org.gradle.tooling.internal.consumer.DefaultModelBuilder$1.run(DefaultModelBuilder.java:64)\\r\\n\\tat org.gradle.tooling.internal.consumer.connection.LazyConsumerActionExecutor.run(LazyConsumerActionExecutor.java:143)\\r\\n\\tat org.gradle.tooling.internal.consumer.connection.CancellableConsumerActionExecutor.run(CancellableConsumerActionExecutor.java:45)\\r\\n\\tat org.gradle.tooling.internal.consumer.connection.ProgressLoggingConsumerActionExecutor.run(ProgressLoggingConsumerActionExecutor.java:61)\\r\\n\\tat org.gradle.tooling.internal.consumer.connection.RethrowingErrorsConsumerActionExecutor.run(RethrowingErrorsConsumerActionExecutor.java:38)\\r\\n\\tat org.gradle.tooling.internal.consumer.async.DefaultAsyncConsumerActionExecutor$1$1.run(DefaultAsyncConsumerActionExecutor.java:67)\\r\\n\\tat org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)\\r\\n\\tat org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:47)\\r\\n\\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)\\r\\n\\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)\\r\\n\\tat java.base/java.lang.Thread.run(Unknown Source)\\r\\nCaused by: org.gradle.cli.CommandLineArgumentException: Unknown command-line option \\u0027--no-daemon\\u0027.\\r\\n\\tat org.gradle.cli.CommandLineParser$BeforeFirstSubCommand.onStartOption(CommandLineParser.java:336)\\r\\n\\tat org.gradle.cli.CommandLineParser.parse(CommandLineParser.java:102)\\r\\n\\tat org.gradle.tooling.internal.provider.ProviderStartParameterConverter.toStartParameter(ProviderStartParameterConverter.java:77)\\r\\n\\t... 18 more\\r\\n"\n' +
    '  }\n' +
    '}\r\n' +
    'java.lang.RuntimeException: java.lang.reflect.InvocationTargetException\r\n' +
    '\tat org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$recursiveFindRpcMethods$0(GenericEndpoint.java:67)\r\n' +
    '\tat org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.notify(GenericEndpoint.java:152)\r\n' +
    '\tat org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleNotification(RemoteEndpoint.java:231)\r\n' +
    '\tat org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:198)\r\n' +
    '\tat org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)\r\n' +
    '\tat org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)\r\n' +
    '\tat org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)\r\n' +
    '\tat java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)\r\n' +
    '\tat java.base/java.util.concurrent.FutureTask.run(Unknown Source)\r\n' +
    '\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)\r\n' +
    '\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)\r\n' +
    '\tat java.base/java.lang.Thread.run(Unknown Source)\r\n' +
    'Caused by: java.lang.reflect.InvocationTargetException\r\n' +
    '\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\r\n' +
    '\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)\r\n' +
    '\tat java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)\r\n' +
    '\tat java.base/java.lang.reflect.Method.invoke(Unknown Source)\r\n' +
    '\tat org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$recursiveFindRpcMethods$0(GenericEndpoint.java:65)\r\n' +
    '\t... 11 more\r\n' +
    'Caused by: java.lang.NullPointerException: Cannot invoke "ch.epfl.scala.bsp4j.TaskId.getId()" because the return value of "ch.epfl.scala.bsp4j.LogMessageParams.getTask()" is null\r\n' +
    '\tat com.microsoft.gradle.bs.importer.GradleBuildClient.onBuildLogMessage(GradleBuildClient.java:78)\r\n' +
    '\t... 16 more\r\n' +
    '\r\n',
  level: 'info',
  timestamp: '2024-07-13 14:14:28.560'
}
{
  message: '[Error - 2:14:28 PM] Jul 13, 2024, 2:14:28 PM Initialization failed \n' +
    "org.eclipse.lsp4j.jsonrpc.ResponseErrorException: org.gradle.tooling.exceptions.UnsupportedBuildArgumentException: Could not fetch model of type 'BuildEnvironment' using connection to Gradle distribution 'https://services.gradle.org/distributions/gradle-8.9-bin.zip'.\n" +
    'Problem with provided build arguments: [--no-daemon]. \n' +
    "Unknown command-line option '--no-daemon'.\n" +
    'Either it is not a valid build option or it is not supported in the target Gradle version.\n' +
    'Not all of the Gradle command line options are supported build arguments.\n' +
    "Examples of supported build arguments: '--info', '-p'.\n" +
    "Examples of unsupported build options: '--daemon', '-?', '-v'.\n" +
    'Please find more information in the javadoc for the BuildLauncher class.\n' +
    "java.util.concurrent.CompletionException: org.eclipse.lsp4j.jsonrpc.ResponseErrorException: org.gradle.tooling.exceptions.UnsupportedBuildArgumentException: Could not fetch model of type 'BuildEnvironment' using connection to Gradle distribution 'https://services.gradle.org/distributions/gradle-8.9-bin.zip'.\n" +
    'Problem with provided build arguments: [--no-daemon]. \n' +
    "Unknown command-line option '--no-daemon'.\n" +
    'Either it is not a valid build option or it is not supported in the target Gradle version.\n' +
    'Not all of the Gradle command line options are supported build arguments.\n' +
    "Examples of supported build arguments: '--info', '-p'.\n" +
    "Examples of unsupported build options: '--daemon', '-?', '-v'.\n" +
    'Please find more information in the javadoc for the BuildLauncher class.\r\n' +
    '\tat java.base/java.util.concurrent.CompletableFuture.reportJoin(Unknown Source)\r\n' +
    '\tat java.base/java.util.concurrent.CompletableFuture.join(Unknown Source)\r\n' +
    '\tat com.microsoft.gradle.bs.importer.GradleBuildServerProjectImporter.importToWorkspace(GradleBuildServerProjectImporter.java:171)\r\n' +
    '\tat org.eclipse.jdt.ls.core.internal.managers.ProjectsManager.importProjects(ProjectsManager.java:164)\r\n' +
    '\tat org.eclipse.jdt.ls.core.internal.managers.ProjectsManager.initializeProjects(ProjectsManager.java:126)\r\n' +
    '\tat org.eclipse.jdt.ls.core.internal.handlers.InitHandler$1.runInWorkspace(InitHandler.java:277)\r\n' +
    '\tat org.eclipse.core.internal.resources.InternalWorkspaceJob.run(InternalWorkspaceJob.java:43)\r\n' +
    '\tat org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)\r\n' +
    "Caused by: org.eclipse.lsp4j.jsonrpc.ResponseErrorException: org.gradle.tooling.exceptions.UnsupportedBuildArgumentException: Could not fetch model of type 'BuildEnvironment' using connection to Gradle distribution 'https://services.gradle.org/distributions/gradle-8.9-bin.zip'.\n" +
    'Problem with provided build arguments: [--no-daemon]. \n' +
    "Unknown command-line option '--no-daemon'.\n" +
    'Either it is not a valid build option or it is not supported in the target Gradle version.\n' +
    'Not all of the Gradle command line options are supported build arguments.\n' +
    "Examples of supported build arguments: '--info', '-p'.\n" +
    "Examples of unsupported build options: '--daemon', '-?', '-v'.\n" +
    'Please find more information in the javadoc for the BuildLauncher class.\r\n' +
    '\tat org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleResponse(RemoteEndpoint.java:220)\r\n' +
    '\tat org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:204)\r\n' +
    '\tat org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)\r\n' +
    '\tat org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)\r\n' +
    '\tat org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)\r\n' +
    '\tat java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)\r\n' +
    '\tat java.base/java.util.concurrent.FutureTask.run(Unknown Source)\r\n' +
    '\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)\r\n' +
    '\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)\r\n' +
    '\tat java.base/java.lang.Thread.run(Unknown Source)\r\n',
  level: 'info',
  timestamp: '2024-07-13 14:14:28.569'
}
```

</details>

--- a/server/src/main/java/com/microsoft/java/bs/core/internal/gradle/GradleApiConnector.java
+++ b/server/src/main/java/com/microsoft/java/bs/core/internal/gradle/GradleApiConnector.java
@@ -67,7 +67,6 @@ public class GradleApiConnector {
   private String getGradleVersion(ProjectConnection connection) {
     BuildEnvironment model = connection
         .model(BuildEnvironment.class)
-        .withArguments("--no-daemon")
         .get();
     return model.getGradle().getGradleVersion();
   }

